src/utils/src/url.js

```diff
@@ -0,0 +1,61 @@
+import url from 'url';
+import qs from 'qs';
+import assign from 'lodash/assign';
+import isArray from 'lodash/isArray';
+import isPlainObject from 'lodash/isPlainObject';
+import isString from 'lodash/isString';
+import isEmpty from 'lodash/isEmpty';
+import { isDefined } from './object';
+
+/**
+ * @private
+ */
+function serialize(obj, options = {}, prefix) {
```

Comment:
@thomasconner similar to promise-queue, can you instead use a third-party package?